src/lib.rs

```diff
@@ -2377,6 +2444,108 @@ mod tests {
         }
     }
 
+    #[test]
+    fn tensor_debug() {
+        let values: Vec<i32> = (1..9).collect();
+        let tests = [
+            (
+                "Tensor<i32> { values: [1, 2, 3, 4, 5, 6, 7, 8], dtype: int32, shape: [8]}",
+                &[8][..],
+                &values,
+            ),
+            (
+                "Tensor<i32> { values: [[1, 2, 3, 4], [5, 6, 7, 8]], dtype: int32, shape: [2, 4]}",
+                &[2, 4],
+                &values,
+            ),
+            (
+                "Tensor<i32> { values: [[[1, 2], [3, 4]], [[5, 6], [7, 8]]], dtype: int32, shape: [2, 2, 2]}",
+                &[2, 2, 2],
+                &values,
+            ),
+        ];
+
+        for &(expected, shape, values) in tests.iter() {
+            let tensor = Tensor::<i32>::new(shape).with_values(values).unwrap();
+            assert_eq!(expected, format!("{tensor:?}"));
+        }
+    }
+
+    fn test_debug_i32_pretty(values: Vec<i32>, shape: &[u64], pretty: &str) {
+        let expected = format!(
+            "Tensor<i32> {{\n    values: {pretty}\n    ],\n    dtype: int32,\n    shape: {shape:?}\n}}"
+        );
+        let tensor = Tensor::<i32>::new(shape).with_values(&values).unwrap();
+        assert_eq!(expected, format!("{tensor:#?}"));
+    }
+
+    #[test]
+    fn tensor_pretty_1_dimension_debug() {
+        let values: Vec<i32> = (1..9).collect();
+        let shape = &[8][..];
+
+        let pretty = "[
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8";
+        test_debug_i32_pretty(values, shape, pretty)
```

Comment:
It'd be a lot easier to verify that these tests are testing the right thing if `pretty` contained the entire string verbatim, i.e. inline `test_debug_i32_pretty` and don't format or concatenate the expected strings in the tests.